includes/class-wc-stripe-order-handler.php

```diff
@@ -40,6 +42,77 @@ public static function get_instance() {
 		return self::$_this;
 	}
 
+	/**
+	 * Disables the ability to edit for orders with uncaptured payment.
+	 *
+	 * @param $editable boolean The current editability of the order.
+	 * @param $order WC_Order The order object.
+	 * @return boolean false if the order has uncaptured payment, true otherwise.
+	 */
+	public function disable_edit_for_uncaptured_orders( $editable, $order ) {
+		try {
+			$intent = $this->get_intent_from_order( $order );
```

Comment:
This is probably a minor thing but what are your thoughts on first checking the orders payment method to confirm it's still a Stripe order before we send off a request to Stripe to fetch an intent object?

Something like:

```
if ( ! WC_Stripe_Helper::payment_method_allows_manual_capture( $order->get_payment_method() ) {
    return $editable;
}
```

I was thinking whether it's possible for a customer to first attempt a purchase with Stripe, it fails, it has an intent ID on it, then they fix it up PayPal. Now each time they view this order, we're sending a request off to Stripe to get the intent object when it's not needed.

Reply:
This is a very good catch. Thanks @mattallan.
Fixed in [878eb9d](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3646/commits/878eb9dae1490418ccfbcc5b9ab3aedb262fe6e4)